src/js/auth.js

```diff
@@ -0,0 +1,41 @@
+const { nanoid } = 'nanoid';
+
+const auth = () => {
+  const getSession = () => {
+    const sessionData = sessionStorage.getItem('session');
+    return sessionData ? JSON.parse(sessionData) : null;
+  };
+
+  const validateEmail = (email) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  const signIn = (email) => {
+    if (!validateEmail(email)) {
+      throw new Error('Invalid email format');
+    }
+
+    const userId = nanoid();
+    const session = {
+      userId,
+      email,
+      created_at: new Date().toISOString(),
+    };
+
+    sessionStorage.setItem('session', JSON.stringify(session));
+    return session;
```

Comment:
Before return the session here, make sure to save this session using the `addLocalStorageUser` created above.